scripts/base/protocols/ldap/main.zeek

```diff
@@ -0,0 +1,365 @@
+  const ports_tcp = { 389/tcp, 3268/tcp } &redef;
+
+  ## UDP ports which should be considered for analysis.
+  const ports_udp = { 389/udp } &redef;
+
+  redef likely_server_ports += { LDAP::ports_tcp, LDAP::ports_udp };
+
+  ## Whether clear text passwords are captured or not.
+  option default_capture_password = F;
+
+  ## Whether to log LDAP search attributes or not.
+  option default_log_search_attributes = F;
+
+  ## Default logging policy hook for LDAP_LOG.
+  global log_policy: Log::PolicyHook;
+
+  ## Default logging policy hook for LDAP_SEARCH_LOG.
+  global log_policy_search: Log::PolicyHook;
+
+  ## LDAP finalization hook.
+  global finalize_ldap: Conn::RemovalHook;
+
+  #############################################################################
+  # This is the format of ldap.log (ldap operations minus search-related)
+  # Each line represents a unique connection+message_id (requests/responses)
+  type MessageInfo: record {
+    # Timestamp for when the event happened.
+    ts: time &log;
+
+    # Unique ID for the connection.
+    uid: string &log;
+
+    # The connection's 4-tuple of endpoint addresses/ports.
+    id: conn_id &log;
+
+    # Message ID
+    message_id: int &log &optional;
+
+    # LDAP version
+    version: int &log &optional;
+
+    # normalized operations (e.g., bind_request and bind_response to "bind")
+    opcode: set[string] &log &optional;
+
+    # Result code(s)
+    results: set[string] &log &optional;
+
+    # result diagnostic message(s)
+    diagnostic_messages: vector of string &log &optional;
+
+    # object(s)
+    objects: vector of string &log &optional;
+
+    # argument(s)
+    arguments: vector of string &log &optional;
+  };
+
+  #############################################################################
+  # This is the format of ldap_search.log (search-related messages only)
+  # Each line represents a unique connection+message_id (requests/responses)
+  type SearchInfo: record {
+    # Timestamp for when the event happened.
+    ts: time &log;
+
+    # Unique ID for the connection.
+    uid: string &log;
+
+    # The connection's 4-tuple of endpoint addresses/ports.
+    id: conn_id &log;
+
+    # Message ID
+    message_id: int &log &optional;
+
+    # sets of search scope and deref alias
+    scopes: set[string] &log &optional;
+    derefs: set[string] &log &optional;
+
+    # base search objects
+    base_objects: vector of string &log &optional;
+
+    # number of results returned
+    result_count: count &log &optional;
+
+    # Result code (s)
+    results: set[string] &log &optional;
+
+    # result diagnostic message(s)
+    diagnostic_messages: vector of string &log &optional;
+
+    #  a string representation of the search filter used in the query
+    filter: string &log &optional;
+
+    # a list of attributes that were returned in the search
+    attributes: vector of string &log &optional;
+  };
+
+  type Info: record {
```

Comment:
Convention is to call these `State`, usually `Info` is meant to be logged. See http or mqtt.